tests/test_config.py

```diff
@@ -0,0 +1,84 @@
+from conftest import upstream_webdav_server
+from flask import url_for
+import os
+import pytest
+import tempfile
+from typing import Any
+
+
+def tmp_certificate(duration):
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
you'd need to do a bit more than self-signed cert, you'd need to create mock CA cert/key, CSR, sign it with CA. Then store CA in CA bundle (actually, replace bundle with just this mock CA).
In this way, it should be tested that any certificate which is not signed by our mock CA is rejected, this includes self-signed certificates.
Let me know if you need help, I could implement this too.

Reply:
I found this:

https://gist.github.com/fntlnz/cf14feb5a46b2eda428e000157447309

it seems about right except that you subject is going to be a person.

By the way, if you want to see a real and almost usable user certificate, go there https://cilogon.org/ login and do "create certificate". Note that this certificate will have insufficient assurance model to access most services, including CSCS services. 
But it's good to see how real grid certificate looks like.
Treat this certificate as real certificate and never share it, since it actually has some, although limited, power.